src/config.ts

```diff
@@ -70,9 +70,9 @@ export async function setShellStartCommand() {
 
   if (shouldAutoShell === "true") {
     // first check the cache
-    const defaultSellCache = localStorage.getItem("terminal.defaultSellCache")
-    if (defaultSellCache && existsSync(defaultSellCache)) {
-      atom.config.set("terminal.shell", defaultSellCache)
+    const defaultShellCache = localStorage.getItem("terminal.defaultSellCache")
+    if (defaultShellCache && existsSync(defaultShellCache)) {
```

Comment:
Use `fs.access` to prevent blocking the main thread

Reply:
Does it really block? We are just awaiting the promise right away. I don't understand how this helps. 🤔 

I did it in:
https://github.com/atom-ide-community/terminal/pull/27/commits/02ad9714d15ffe09c13b5a4e982946da262cefc8

Reply:
> Does it really block?

Yes, and for a slow hard drive it can be multiple seconds. For `existsSync` node runs a separate process to check the file system and it blocks the main thread until that process finishes.